<div style="display: flex; align-items: center;">
    <div style="flex: 1; font-family: Arial, sans-serif; text-align: center;">
        <h1>UNIVERSIDAD CENTRAL DEL ECUADOR</h1>
        <h2>FACULTAD DE CIENCIAS ECONÓMICAS</h2>
        <h3>ESCUELA DE INGENIERÍA ESTADÍSTICA</h3>
        <h4>SURVIVAL ANALYSIS</h4>
    </div>
    <div style="flex: 1; text-align: right;">
        <img src="ACTUARIAL/Analisis.png" alt="Descripción de la imagen">
    </div>
</div>

<div style="background-color: #AEEEEE; color: black; padding: 20px; text-align: center;">
    <h1>ANÁLISIS DE SUPERVIVENCIA</h1>
</div>

**¿Qué es el Análisis de Supervivencia?**


- El análisis de supervivencia examina y modela el tiempo que tardan en producirse lo que se denomina “eventos aleatorios”.
- El resultado que se recoge es:
  1. El tiempo que acontece el suceso.
  2. Tiempo transcurrido hasta la finalización del período de seguimiento del experimento.


<img src="ACTUARIAL/Historia.png" alt="Descripción de la imagen" style="width: 100%; height: auto;">


<img src="ACTUARIAL/Evento.png" alt="Descripción de la imagen" style="width: 100%; height: auto;">

<div style="background-color: #AEEEEE; color: black; padding: 20px; text-align: center;">
    <h1>DATOS CENSURADOS O INCOMPLETOS</h1>
</div>

<img src="ACTUARIAL/Datos.png" alt="Descripción de la imagen" style="width: 70%; height: auto;">



**Representación Gráfica**

<img src="ACTUARIAL/Representación.png" alt="Descripción de la imagen" style="width: 70%; height: auto;">


- Se centra en estudiar la función de distribución del tiempo de supervivencia o tiempo hasta que ocurre el evento de interés.
- Tiene en cuenta los eventos que se vuelven inobservables después de un cierto punto de tiempo o no ocurren durante el período de monitoreo.



<div style="background-color: #AEEEEE; color: black; padding: 20px; text-align: center;">
    <h1>FUNCIONES DE SUPERVIVENCIA</h1>
</div>

Si 𝑓(𝑡) denota la función de densidad de probabilidad para la variable aleatoria 𝑇, tiempo de supervivencia, y 𝐹(𝑡) a la correspondiente función de distribución; entonces, se define la función de supervivencia 𝑆(𝑡) como la probabilidad de sobrevivir al menos hasta el instante 𝑡:

<center>𝑆(𝑡)=𝑃(𝑇>𝑡)=1−𝐹(𝑡)<center>



<div style="background-color: white; color: black; padding: 15px; text-align: center;">
    <h1>Función Hazard</h1>
</div>
<img src="ACTUARIAL/Hazard.png" alt="Descripción de la imagen" style="width: 100%; height: auto;">


<div style="background-color: white; color: black; padding: 15px; text-align: center;">
    <h1>Función Hazard Acumulada</h1>
</div>

<center>A mayor tiempo transcurrido, mayor probabilidad de que ocurra el evento de interés:<center>
    
    
<center>𝐻(𝑡)=−log⁡[𝑆(𝑡)]<center>

<div style="background-color: #AEEEEE; color: black; padding: 20px; text-align: center;">
    <h1>MODELOS DE SUPERVIVENCIA</h1>
</div>
<img src="ACTUARIAL/Modelos.png" alt="Descripción de la imagen" style="width: 100%; height: auto;">


<div style="background-color: white; color: black; padding: 15px; text-align: center;">
    <h1>Cox Proportional Hazard</h1>
</div>

- Modelo de regression.
- Investigaciones médicas: asociación entre el tiempo de sobrevivencia de los pacientes y una o más variables predictoras.
- Funciona con variables predictoras cuantitativas y categóricas.

<img src="ACTUARIAL/Cox1.png" alt="Descripción de la imagen" style="width: 40%; height: auto;">

<img src="ACTUARIAL/Cox2.png" alt="Descripción de la imagen" style="width: 70%; height: auto;">

<img src="ACTUARIAL/Cox3.png" alt="Descripción de la imagen" style="width: 70%; height: auto;">

<img src="ACTUARIAL/Cox4.png" alt="Descripción de la imagen" style="width: 70%; height: auto;">

<center>Para facilidad de interpretación:<center>

<center>log⁡(ℎ(𝑡,𝑋))=log⁡[(ℎ(𝑡))+𝑋𝛽]<center>

<div style="background-color: #AEEEEE; color: black; padding: 20px; text-align: center;">
    <h1>APLICACIÓN MODELO DE COX</h1>
</div>


Los datos siguientes provienen de un estudio de supervivencia en pacientes con cáncer de pulmón avanzado del North Central Cancer Treatment Group. A cada paciente se le pasan diferentes test para medir su capacidad para desempeñarse habitualmente, así como variables asociadas con su alimentación. El conjunto de variable consideradas son: inst (código de la institución), time (tiempo de supervivencia en días), status (estado de la censura 1=censurado, 2=muerto), age (edad en años), sex (masculino=1 femenino=2), ph.ecog (puntuación del test ECOG (0=buen estado 5=muerto)), ph.karno (puntuación del test de Karnofsky (malo=0-bueno=100)), pat.karno (test de Karnofsky estandarizado por paciente), meal.cal (consumo de calorias), wt.loss (peso pérdido en los últimos seis meses). Se desea estudiar la supervivencia de los pacientes en función de las variables registradas.

In [1]:
def toggle_cell_code(button_id):
    """
    Adds a button to toggle (show/hide) the code cell but not the output. 
    
    Parameters
    ----------
    button_id : str
        An identifier for cells that will hide/show when button is pressed.
    """
    
    from IPython.display import display_html
    my_html = '''
    <button type="button" id="%s" onclick="code_toggle('%s')">Codigo</button>
    <script>
    function code_toggle(my_id) {
        // get the parent element for the cell code and output
        var p = $("#"+my_id);
        if (p.length==0) return;
        while (!p.hasClass("cell")) {
            p = p.parent();
            if (p.prop("tagName") =="body") return;
        }
    // get the cell code and toggle its value
    var cell_code = p.find(".input");
    cell_code.toggle();
    }
    </script>
    ''' %(button_id, button_id)
    return display_html(my_html, raw=True)
toggle_cell_code("cero")

Codigo

<div style="background-color: white; color: black; padding: 20px; text-align: center;">
    <h1>Análisis Inicial</h1>
</div>

In [4]:
toggle_cell_code("uno")
import rpy2.robjects as robjects
import warnings
warnings.filterwarnings('ignore')
# Código R a ejecutar
r_code = '''
options(warn = -1)
library(ggplot2)
library(dplyr)
library(ggpubr)
library(survival)
library(survminer)
library(prodlim)
library(pec)
library(descr)
library(forcats)
#Importar Datos
data("lung")
print(head(lung))
#
'''
# Ejecutar el código R en Python
robjects.r(r_code)

Codigo

  inst time status age  sex ph.ecog ph.karno pat.karno meal.cal wt.loss
1    3  306      2  74 Male       1       90       100     1175      NA
2    3  455      2  68 Male       0       90        90     1225      15
3    3 1010      1  56 Male       0       90        90       NA      15
4    5  210      2  57 Male       1       90        60     1150      11
5    1  883      2  60 Male       0      100        90       NA       0
6   12 1022      1  74 Male       1       50        80      513       0
  censurado
1         1
2         1
3         0
4         1
5         1
6         0


In [3]:
toggle_cell_code("dos")
# Código R a ejecutar
r_code = '''
lung <- lung %>% 
  mutate(censurado = status - 1,
         sex = fct_recode(as.factor(sex),"Male" = "1", "Female" = "2"))
# Estimación
fit <- survfit(Surv(time,censurado) ~ 1, data = lung)
print(fit)
'''
# Ejecutar el código R en Python
robjects.r(r_code)


Codigo

Call: survfit(formula = Surv(time, censurado) ~ 1, data = lung)

       n events median 0.95LCL 0.95UCL
[1,] 228    165    310     285     363


Se puede observar el resumen de la mediana  de supervivencia del todo el conjunto de datos. La mediana de supervivencia se sitúa en 310 días con un intervalo de confianza del 95% entre 285 y 363 días. A continuación se realiza el análisis considerando la variable ‘sexo’ que indica el "tratamiento" del sujeto. Representamos las curvas de supervivencia, la mediana de supervivencia, y el pvalor del contraste de comparación de funciones hazard.


In [5]:
toggle_cell_code("tres")
# Gráfica de la función de supervivencia y mediana de supervivencia
# Código R a ejecutar
r_code = '''
# Puedes ajustar las opciones según tus necesidades
#my_plot <- ggsurvplot(fit, data = lung, 
           palette = "purple", 
           conf.int = FALSE, 
           surv.median.line = "hv")
# Estimación
fit <- survfit(Surv(time,censurado) ~ sex, data = lung)
fit
# Gráfica de la función de supervivencia e IC
#ggsurvplot(fit, data = lung, 
           conf.int = TRUE, 
           conf.int.style = "step", 
           surv.median.line = "hv", 
           pval = TRUE)
'''

# Ejecutar el código R en Python
#result = robjects.r(r_code)
#print(result)


Codigo

**Función de Supervivencia Global y por Sexo**
<table>
  <tr>
    <td>
      <img src="ACTUARIAL/Grafico1.png" alt="Descripción de la imagen 1" style="width: 80%;">
    </td>
    <td>
      <img src="ACTUARIAL/Grafico2.png" alt="Descripción de la imagen 2" style="width: 80%;">
    </td>
  </tr>
</table>

Podemos ver que si existe  diferencias entre las curvas de supervivencia para los dos grupos considerados a nivel de sexo (p-valor superior a 0.05). 
 

<div style="background-color: white; color: black; padding: 20px; text-align: center;">
    <h1>Modelo de Supervivencia</h1>
</div>


In [9]:
toggle_cell_code("cuatro")
r_code = '''
# Ajuste del modelo
fitcom <- coxph(Surv(time, status) ~ sex + age + wt.loss, data = lung)
# Resumen del modelo ajustado
print(fitcom)
# Gráfico de la relevancia de los coeficientes
#ggforest(fitcom)
#ggadjustedcurves(fitcom, data=lung, variable = "sex", palette = "lancet")
# Edad
# Variable  artificial
#lung$agenew <- cut(lung$age,quantile(lung$age,na.rm = TRUE))
# Gráfico de supervivencia para sex
#ggadjustedcurves(fitcom, data=lung, variable = "agenew",  palette = "lancet")
'''
# Ejecutar el código R en Python
robjects.r(r_code)

Codigo

Call:
coxph(formula = Surv(time, status) ~ sex + age + wt.loss, data = lung)

                coef  exp(coef)   se(coef)      z      p
sexFemale -0.5210319  0.5939074  0.1743541 -2.988 0.0028
age        0.0200882  1.0202913  0.0096644  2.079 0.0377
wt.loss    0.0007596  1.0007599  0.0061934  0.123 0.9024

Likelihood ratio test=14.67  on 3 df, p=0.002122
n= 214, number of events= 152 
   (14 observations deleted due to missingness)


Una vez ejecutado el Cox Proportional Hazard se puede notar en primer lugar que se está en presencia de un buen modelo de regresión, donde las variables sexo y edad son las que explican de mejor forma la ocurrencia o no del evento de estudio (muerte del individuo).
Analizando al sexo, se puede concluir que, si el individuo es mujer, tiene un 59.4% menos probabilidad de fallecer respecto a los hombres. De esta forma se evidencia que las mujeres tienen mejor pronóstico de vida respecto a los hombres
Por el lado de la edad, por cada año adicional que tenga la persona, su probabilidad de fallecer va a incrementarse en un 102%.
El test global indica que el modelo construido es útil para explicar la supervivencia de los sujetos. Veamos el análisis gráfico de los coeficientes del modelo:

**Gráfico de la Relevancia de los Coeficientes**
<img src="ACTUARIAL/Gráfico3.png" alt="Descripción de la imagen" style="width: 80%; height: auto;">

**Curvas de Supervivencia por Sexo y Edad**

<table>
  <tr>
    <td>
      <img src="ACTUARIAL/Grafico4.png" alt="Descripción de la imagen 1" style="width: 80%;">
    </td>
    <td>
      <img src="ACTUARIAL/Grafico5.png" alt="Descripción de la imagen 2" style="width: 80%;">
    </td>
  </tr>
</table>
 

### Tests de riesgos proporcionales

Sostiene que el cociente de las tasas de riesgo entre dos grupos comparables es constante y no cambia con el tiempo. Si esta hipótesis se cumple, es apropiado utilizar el modelo de riesgos proporcionales de Cox para analizar los datos de supervivencia.

In [22]:
toggle_cell_code("cuatro_0")
r_code = '''
ftest <- cox.zph(fitcom)
print(ftest)
'''
# Ejecutar el código R en Python
robjects.r(r_code)

Codigo

         chisq df    p
sex     2.5489  1 0.11
age     0.5077  1 0.48
wt.loss 0.0144  1 0.90
GLOBAL  3.0051  3 0.39


Dado que el pvalor asociado del test resulta no significativo no podemos descartar la hipótesis de riesgos proporcionales asociados con la variable sexo, age y peso.

### Observaciones Influyentes

In [25]:
toggle_cell_code("cuatro_1")
r_code = '''
# Observaciones influyentes
#ggcoxdiagnostics(fitcomp, type = "deviance", linear.predictions = FALSE)
'''
# Ejecutar el código R en Python
robjects.r(r_code)

Codigo

<rpy2.rinterface_lib.sexp.NULLType object at 0x000002394DDA11C0> [RTYPES.NILSXP]

<img src="ACTUARIAL/Grafico9.png" alt="Descripción de la imagen" style="width: 70%; height: auto;">

### No Linealidad


In [27]:
toggle_cell_code("cuatro_2")
r_code = '''
# No linealidad
#ggcoxdiagnostics(fitcomp, type = "schoenfeld",ox.scale = "time")
'''
# Ejecutar el código R en Python
robjects.r(r_code)

Codigo

<rpy2.rinterface_lib.sexp.NULLType object at 0x000002394DDA11C0> [RTYPES.NILSXP]

<img src="ACTUARIAL/Grafico10.png" alt="Descripción de la imagen" style="width: 70%; height: auto;">

## Puntos de Corte de la Variable Edad y Peso

In [11]:
toggle_cell_code("cinco")
r_code = '''
# Puntos de corte
corte <- surv_cutpoint(lung, time = "time", 
              event = "status", 
              variables = c("age","wt.loss"))
print(corte)
'''
# Ejecutar el código R en Python
robjects.r(r_code)

Codigo

        cutpoint statistic
age           70  2.013619
wt.loss        9  1.756750


El valor para edad se sitúa en los 70 años mientras que para la pérdida de peso es de 9kg. 

### Representación de las Curvas de Supervivencia de la Edad y Peso usando Puntos de Corte

In [18]:
toggle_cell_code("seis")
r_code = '''
# Variable  artificial edad
lung$agenew <- ifelse(lung$age>=corte$age$estimate,"+70","-70")
lung$wtnew <- ifelse(lung$wt.loss>=corte$wt.loss$estimate,"+9","-9")
# Gráfico de supervivencia para edad
# ggadjustedcurves(fitcom, data=lung,variable = "agenew", palette = "lancet")
# Gráfico de supervivencia para pérdida de peso
#ggadjustedcurves(fitcom, data=lung, variable = "wtnew", palette = "lancet")
'''
# Ejecutar el código R en Python
robjects.r(r_code)

Codigo

<table>
  <tr>
    <td>
      <img src="ACTUARIAL/Grafico6.png" alt="Descripción de la imagen 1" style="width: 80%;">
    </td>
    <td>
      <img src="ACTUARIAL/Grafico7.png" alt="Descripción de la imagen 2" style="width: 80%;">
    </td>
  </tr>
</table>

## Modelo Ajustado a las Nuevas Variables

In [20]:
toggle_cell_code("siete")
r_code = '''
# Ajuste del modelo
fitcomp2 <- coxph(Surv(time, status) ~ sex + agenew + wtnew, data = lung)
print(fitcomp2)
## Gráfico conjunto de todos los fatores
#ggsurvplot_facet(fitcomp2, lung, facet.by = "wtnew",palette = "lancet", pval = TRUE)
'''
# Ejecutar el código R en Python
robjects.r(r_code)

Codigo

Call:
coxph(formula = Surv(time, status) ~ sex + agenew + wtnew, data = lung)

             coef exp(coef) se(coef)      z       p
sexFemale -0.5160    0.5969   0.1744 -2.960 0.00308
agenew+70  0.3754    1.4555   0.1800  2.086 0.03702
wtnew+9    0.2370    1.2675   0.1628  1.456 0.14549

Likelihood ratio test=16.34  on 3 df, p=0.0009675
n= 214, number of events= 152 
   (14 observations deleted due to missingness)


### Representación de la Curva de Supervivencia 

<img src="ACTUARIAL/Grafico8.png" alt="Descripción de la imagen" style="width: 70%; height: auto;">

### Test de Riesgos Proporcionales

In [31]:
toggle_cell_code("siete_0")
r_code = '''
# Tests de riesgos proporcionales
ftest <- cox.zph(fitcomp2)
print(ftest)
'''
# Ejecutar el código R en Python
robjects.r(r_code)




Codigo

       chisq df    p
sex    2.617  1 0.11
agenew 0.138  1 0.71
wtnew  0.187  1 0.67
GLOBAL 2.916  3 0.40


Dado que el pvalor asociado del test resulta no significativo no podemos descartar la hipótesis de riesgos proporcionales asociados con la variable sexo, agenew y wtnew.

### Gráfica de las Observaciones Influyentes y No Linealidad



In [33]:
toggle_cell_code("siete_1")
r_code = '''
# Observaciones influyentes
#ggcoxdiagnostics(fitcomp2, type = "deviance",linear.predictions = FALSE)

'''
# Ejecutar el código R en Python
robjects.r(r_code)

Codigo

<rpy2.rinterface_lib.sexp.NULLType object at 0x000002394DDA11C0> [RTYPES.NILSXP]

<table>
  <tr>
    <td>
      <img src="ACTUARIAL/Grafico11.png" alt="Descripción de la imagen 1" style="width: 80%;">
    </td>
    <td>
      <img src="ACTUARIAL/Grafico12.png" alt="Descripción de la imagen 2" style="width: 80%;">
    </td>
  </tr>
</table>

De forma gráfica no se observa la presencia de observaciones influyentes y en el caso de la no linealidad en este modelo se ve que mejora en comparación al anterior.

<div style="background-color: #AEEEEE; color: black; padding: 20px; text-align: center;">
    <h1>CONCLUSIONES</h1>
</div>

- El modelo de Cox permite el análisis y la estimación de la tasa de riesgo a lo largo del tiempo, incluso en presencia de datos censurados. Esto lo hace especialmente útil en estudios en los que se siguen a los sujetos durante un período prolongado y algunos de ellos no experimentan el evento de interés o abandonan el estudio antes de que ocurra el evento.

- El modelo permite identificar qué factores influyen significativamente en la supervivencia y cuánto afectan a la tasa de riesgo relativo. Esto permite a los investigadores entender mejor la influencia de diferentes variables en el tiempo hasta que ocurra el evento, lo que puede tener importantes implicaciones clínicas y epidemiológicas. 

<div style="background-color: #AEEEEE; color: black; padding: 20px; text-align: center;">
    <h1>REFERENCIAS BIBLIOGRÁFICAS</h1>
</div>

<img src="ACTUARIAL/Referencias.png" alt="Descripción de la imagen" style="width: 80%; height: auto;">